Sources/MongoSwift/BSON/BSONDecoder.swift

```diff
@@ -534,7 +534,7 @@ private struct _BSONKeyedDecodingContainer<K: CodingKey> : KeyedDecodingContaine
         let entry = try getValue(forKey: key)
         return try self.decoder.with(pushedKey: key) {
             let value = try decoder.unbox(entry, as: type)
-            guard !(value is BSONNull) else {
+            guard !(value is BSONNull) || type == BSONNull.self else {
```

Comment:
what necessitated this change?

Reply:
I noticed that I never added `BSONNull` to `AllBSONTypes` (shame), and I was under the impression that I used `AllBSONTypes` for testing the changes as part of this ticket. That's not _technically_ true, though, so I can revert them if it clouds this PR.

The exact change referenced here is to allow the decoder to decode `BSONNull` if it's a member of a struct/class. It's a very uncommon case but is still a bug (which was detected by adding `BSONull` to `AllBSONTypes`).

It's encountered when someone tries decoding to a type like this:
```swift
struct NonsensicalType {
     let a: BSONull
}
```

Reply:
I'm fine with leaving it, I was just curious about what case would case would lead to this.